pkg/applicationserver/io/packages/alcsync/v1/commands.go

```diff
@@ -0,0 +1,51 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alcsyncv1
+
+import (
+	"context"
+	"time"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+)
+
+var errUnknownCommand = errors.DefineNotFound("unknown_command", "unknown command")
+
+// Converts GPS time to UTC time with second precision.
+func convertToUTCTime(gpsTime uint32) time.Time {
+	gpsEpoch := time.Date(1980, 1, 6, 0, 0, 0, 0, time.UTC)
```

Comment:
Please note that this does not take the leap seconds into consideration and will output the wrong timestamps. Please use `pkg/gpstime`.